### PR TITLE
Fix AssertEx silliness

### PR DESCRIPTION
**Pull request by jasonmalinowski**
The fixing of the AssertEx.ThrowsArgument* helpers is hilarious.

*Reviewers:* @dotnet/roslyn-compiler, @dotnet/roslyn-ide

### CODE REVIEW

**Comment by Pilchie** [comment]
:+1:

**Comment by TyOverby** [comment]
:shipit:

File: src/Test/Utilities/Shared/Assert/AssertEx.cs [hunk 1]
```diff
@@ -331,55 +331,14 @@ public static void NotNull<T>(T @object, string message = null)
...
         {
-            try
-            {
-                del();
-            }
-            catch (ArgumentNullException e)
-            {
-                Assert.Equal(parameterName, e.ParamName);
-            }
+            var exception = Assert.Throws<ArgumentNullException>(del);
```
Comment (davkean) [inline src/Test/Utilities/Shared/Assert/AssertEx.cs]: xunit already has Assert.Throws<ArgumentNullException>(parameterName, Action) that does this.

File: src/Test/Utilities/Shared/Assert/AssertEx.cs [hunk 2]
```diff
@@ -331,55 +331,14 @@ public static void NotNull<T>(T @object, string message = null)
...
-                    return (T)ex;
-                }
-
-                // We got some other type. We know that type != typeof(T), and so we'll use Assert.Equal since Xunit
-                // will give a nice Expected/Actual output for this
-                Assert.Equal(typeof(T), type);
-            }
-
-            throw new Exception("No exception was thrown.");
+            var exception = Assert.Throws<ArgumentException>(del);
```
Comment (davkean) [inline src/Test/Utilities/Shared/Assert/AssertEx.cs]: Ditto

**Comment by jasonmalinowski** [comment]
It turns out fixing the Throws* helpers actually exposed a scripting test bug. @dotnet/roslyn-interactive now gets to review this too! There's a separate commit with the actual change the tests.

**Comment by gafter** [comment]
:+1:

**Comment by TyOverby** [comment]
@jasonmalinowski What changed about the helpers that exposed the bug?

**Comment by jasonmalinowski** [comment]
@TyOverby: Assert.Throws() should fail the test if the code _doesn't_ throw an exception. Our helpers would let the tests pass. So we had tests asserting that some code would throw exceptions but it didn't in reality.

**Comment by jasonmalinowski** [comment]
@dotnet/roslyn-interactive, can I get a review of 5b01817235c071849ac02e62bc982a5a7911a67f specifically?

**Comment by KevinH-MS** [comment]
5b01817 LGTM...I can see no reason why ```ScriptOptions.AddReferences``` should throw in this case.

**Comment by jasonmalinowski** [comment]
Thanks all! I agree with @KevinH-MS's assessment.

**Pull request merged**